pinot-core/src/main/java/org/apache/pinot/core/periodictask/PeriodicTask.java

```diff
@@ -59,6 +59,12 @@
   @Override
   void run();
 
+  /**
+   * Execute the task once. This method will calls the {@link #run} method.
+   * @param filter An implementation specific string that may dictate how the task will be run. null by default.
+   */
+  void run(String filter);
```

Comment:
Can we use Context object instead of a filter string? By using Context, it'll be easier to maintain and more flexible to extend.

Reply:
Looking into this...